src/Parser/inc/Parser.h

```diff
@@ -18,32 +20,61 @@ namespace calculator
         float a;
         float b;
         char operation;
-        bool valid = true;
+        std::string fullExpression;
+        int placementIndex = 0;
+        bool validExpression = false;
+        bool hasParenthesis = false;
+    };
+
+    struct ExpressionUnit
+    {
+        float number;
+        char  operation;
+        bool  valid = false;
     };
 
     class Parser
     {
         public:
-            /**
-             * @brief uses regex to ensure value is a float
-             * @return isValid if it is a float
-            */
-            bool validateFloat(const std::string &input);
-
             /**
              * @brief the main parsing function that puts all the peices together and returns an expression
              * @return an Expression object, along with whether it is valid or not
             */
-            Expression parseFullEquation(const std::string &fullEquation);
+            Expression breakDownEquation(const std::shared_ptr<std::vector<ExpressionUnit>> &equationVector);
 
             /**
              * @brief takes in input
              * @returns string of input entered
             */
             std::string getUserInput();
 
+            /**
+             * @brief creates a vector out of a string
+             * @returns vector split into each unit of expression
+            */
+            std::pair <std::shared_ptr<std::vector<ExpressionUnit>>,bool> createVector(const std::string &fullEquation);
+
+            /**
+             * @brief returns the original equation string
+             * @returns the unedited string input by user
+            */
+            std::string getOriginalEquation();
         private:
-            char Operations[6] = {'+', '-', '/', '%', '*', 'x'};
+            std::string editedEquation = "";

```

Comment:
private members should have a _ suffix e.g., `std::string editedEquation_ = "";`